scripts/generate-notices.py

```
@@ -0,0 +1,235 @@
+    subprocess.run(
+        [pip, "install", "--quiet", "--upgrade", *[f"{n}=={v}" for n, v in deps]],
+        check=True,
+    )
+    raw = subprocess.check_output(
+        [pip_licenses, "--packages", *[n for n, _ in deps],
+         "--with-license-file", "--with-urls", "--format=json", "--no-license-path"],
+        text=True,
+    )
+    entries = sorted(json.loads(raw), key=lambda e: e["Name"].lower())
+
+    def src(e):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `src`

(ANN202)